uaclient/cli.py

```diff
@@ -1619,6 +1620,64 @@ def action_refresh(args, *, cfg: config.UAConfig):
     return 0
 
 
+def is_contract_changed(cfg: config.UAConfig) -> bool:
+    orig_token = cfg.machine_token
+    orig_entitlements = cfg.entitlements
+    machine_token = orig_token["machineToken"]
+    contract_id = orig_token["machineTokenInfo"]["contractInfo"]["id"]
+    contract_client = contract.UAContractClient(cfg)
+    resp = contract_client.get_updated_contract_info(
+        machine_token, contract_id
+    )
+    try:
+        resp_ = json.dumps(resp, cls=util.DatetimeAwareJSONEncoder)
+        resp = json.loads(resp_, cls=util.DatetimeAwareJSONDecoder)
```

Comment:
Why are encoding the response to json and immediately decoding it back ? Is it because of the dates ?

Reply:
+1 I had the same question

Reply:
Yes, it is because of the date part.